### DISCUSSION TITLE
Outline Headings and Titles

### DISCUSSION THREAD

**Discussion by EggerGideon** [typst/typst#3341] [Q&A] [answered]
In the screenshot below you can see the structure of my file. Why is it sometimes buggy like in 1.1.3.1? Is there a solution? Sometimes it works, as 1.1.1.2 shows. - *Solved see answer below*

Another question would be if there was an option to remove heading 1 but still show it as a "top heading" and start with design as 1 instead. - *Solved see marked answer*

For example:
Now:
```
1. Heading
1.1. Topic
1.1.1. Subtopic
...
```

Desired result:
```
Heading
1. Topic
1.1. Subtopic
...
```

Thanks for the help. Don't hesitate to request more information.
(I have searched for both problems in the web, but nothing was found)

![image](https://github.com/typst/typst/assets/91525575/89e9d8e7-5858-4382-9791-4a41ab63debf)

**Comment by Julian-Wassmann** [comment]
Hi!

Can you please provide a code example which reproduces your issue? Either by sharing a project via share link or by posting the code here.
Looks like you changed some styling, which is causing this issue. We need more information about this.

**Reply by EggerGideon** [reply]
Thanks for the quick reply:

Here is the code:
```
#show outline.entry.where(
  level: 1
): it => {
  v(12pt, weak: true)
  strong(it)
}
#outline(indent: 1em, depth: 4)
```

the template.typ file does not change any setting related to the outline.

**Reply by Julian-Wassmann** [reply]
This doesn't explain why the issue is happening. For further investigation it is necessary that you share code which reproduces the issue. So not only the styling rules, but also the document content.
If you have some time, please provide a minimum working example (which contains your issue).

**Reply by EggerGideon** [reply]
Ok, I can understand and now even more ... when you gave me the hint to provide you even more context to replicate the error.
That was the solution.

Should debug more often with opening a blank project do replicate errors ...

Check out this simple error that I explain in the additional answer.

Edit: And thank you very much for your help!

**Comment by EggerGideon** [comment]
To the question/"bug" shown in the screenshot:

Due to a circumstance I had to add a backslash next to a few subtopics. But this also changes the behavior in the “Outline” which produces this “error”.

With '\' in Text:
```
= Topic\
```

Outline:
```
1. Topic
.................................................... 1
```



Without '\' in Text:
```
= Topic
```

Outline:
```
1. Topic ........................................... 1
```

**Comment by Julian-Wassmann** [comment, marked as answer]
Answer to your second question:

Use the `numbering` parameter of `heading` and don't input a string (numbering pattern) but a numbering function. This way it is possible to add some custom logic to discard the chapter number. Have a look into the [numbering docs](https://typst.app/docs/reference/model/numbering/) for an explanation. The change is applied not only to numbering in headings, but also in heading references and the outline.
```typst
#set heading(numbering: (..nums) => {
  let sequence = nums.pos()
  // discard first entry (chapter number)
  let _ = sequence.remove(0)
  if sequence.len() > 0 {
    numbering("1.", ..sequence)
  }
})
```

**Reply by EggerGideon** [reply]
Worked. Thank you very much for your help!
